x/gamm/types/msgs_test.go

```diff
@@ -867,3 +872,148 @@ func TestMsgExitSwapShareAmountIn(t *testing.T) {
+			expectedRevokeSignByteMsg:  `{"account_number":"1","chain_id":"foo","fee":{"amount":[],"gas":"0"},"memo":"memo","msgs":[{"type":"cosmos-sdk/MsgRevoke","value":{"grantee":"cosmos1def","granter":"cosmos1abc","msg_type_url":"/osmosis.gamm.v1beta1.MsgSwapExactAmountIn"}}],"sequence":"1","timeout_height":"1"}`,
+			expectedExecStrSignByteMsg: fmt.Sprintf(`{"account_number":"1","chain_id":"foo","fee":{"amount":[],"gas":"0"},"memo":"memo","msgs":[{"type":"cosmos-sdk/MsgExec","value":{"grantee":"cosmos1def","msgs":[{"type":"osmosis/gamm/swap-exact-amount-in","value":{"routes":[{"token_out_denom":"test"},{"pool_id":"1","token_out_denom":"test2"}],"sender":"%s","token_in":{"amount":"1","denom":"stake"},"token_out_min_amount":"1"}}]}}],"sequence":"1","timeout_height":"1"}`, addr1),
+			gammMsg: &MsgSwapExactAmountIn{
+				Sender: addr1,
+				Routes: []SwapAmountInRoute{{
+					PoolId:        0,
+					TokenOutDenom: "test",
+				}, {
+					PoolId:        1,
+					TokenOutDenom: "test2",
+				}},
+				TokenIn:           coin,
+				TokenOutMinAmount: sdk.NewInt(1),
+			},
+		},
+		{
+			expectedGrantSignByteMsg:   `{"account_number":"1","chain_id":"foo","fee":{"amount":[],"gas":"0"},"memo":"memo","msgs":[{"type":"cosmos-sdk/MsgGrant","value":{"grant":{"authorization":{"type":"cosmos-sdk/GenericAuthorization","value":{"msg":"/osmosis.gamm.v1beta1.MsgSwapExactAmountOut"}},"expiration":"0001-01-01T02:01:01.000000001Z"},"grantee":"cosmos1def","granter":"cosmos1abc"}}],"sequence":"1","timeout_height":"1"}`,
+			expectedRevokeSignByteMsg:  `{"account_number":"1","chain_id":"foo","fee":{"amount":[],"gas":"0"},"memo":"memo","msgs":[{"type":"cosmos-sdk/MsgRevoke","value":{"grantee":"cosmos1def","granter":"cosmos1abc","msg_type_url":"/osmosis.gamm.v1beta1.MsgSwapExactAmountOut"}}],"sequence":"1","timeout_height":"1"}`,
+			expectedExecStrSignByteMsg: fmt.Sprintf(`{"account_number":"1","chain_id":"foo","fee":{"amount":[],"gas":"0"},"memo":"memo","msgs":[{"type":"cosmos-sdk/MsgExec","value":{"grantee":"cosmos1def","msgs":[{"type":"osmosis/gamm/swap-exact-amount-out","value":{"routes":[{"token_in_denom":"test"},{"pool_id":"1","token_in_denom":"test2"}],"sender":"%s","token_in_max_amount":"1","token_out":{"amount":"1","denom":"stake"}}}]}}],"sequence":"1","timeout_height":"1"}`, addr1),
+			gammMsg: &MsgSwapExactAmountOut{
+				Sender: addr1,
+				Routes: []SwapAmountOutRoute{{
+					PoolId:       0,
+					TokenInDenom: "test",
+				}, {
+					PoolId:       1,
+					TokenInDenom: "test2",
+				}},
+				TokenOut:         coin,
+				TokenInMaxAmount: sdk.NewInt(1),
+			},
+		},
+	}
+	for _, tc := range testCases {
+		// Authz: Grant Msg
+		typeURL := sdk.MsgTypeURL(tc.gammMsg)
+		grant, err := authz.NewGrant(someDate, authz.NewGenericAuthorization(typeURL), someDate.Add(time.Hour))
+		require.NoError(t, err)
+		msgGrant := &authz.MsgGrant{Granter: "cosmos1abc", Grantee: "cosmos1def", Grant: grant}
```

Comment:
can we extract `cosmos1abc` and `cosmos1def` to constants instead of hardcoding in multiple places, please?